src/app/core/layers/layersstore.js

```diff
@@ -288,28 +288,37 @@ proto.createLayersTree = function(groupName, options={}) {
   const tocLayersId = this.getLayers({BASELAYER:false}).map(layer=>layer.getId());
   let layerstree = [];
   if (_layerstree) {
-    if (full === true) return this.state.layerstree;
-    else {
+    if (full === true) {
+      return this.state.layerstree;
+    } else {
       let traverse = (obj, newobj) => {
         obj.forEach(layer => {
-          let lightlayer = {};
-          if (layer.id !== null && layer.id !== undefined) {
-            if (tocLayersId.find(toclayerId => toclayerId === layer.id)) {
-              lightlayer.id = layer.id;
-            } else lightlayer = null;
+          let lightlayer = null;
+
+          // case TOC has layer ID
+          if (null !== layer.id && "undefined" !== typeof layer.id && tocLayersId.find(id => id === layer.id)) {
+            lightlayer = ({ ...lightlayer, id: layer.id });
           }
+
           // case group
-          if (layer.nodes !== null && layer.nodes !== undefined) {
-            lightlayer.title = layer.name;
-            lightlayer.groupId = uniqueId();
-            lightlayer.nodes = [];
-            lightlayer.checked = layer.checked;
-            lightlayer.mutually_exclusive = layer["mutually-exclusive"];
-            traverse(layer.nodes, lightlayer.nodes)
+          if (null !== layer.nodes && "undefined" !== typeof layer.nodes) {
+            lightlayer = ({
+              ...lightlayer,
```

Comment:
Feel free to revert to my last commit if you think we'll never need to store the `id: layer.id` property within the second if as well (although that was the behavior before this pull)